src/datasources/cow-swap/swaps-api.factory.ts

```diff
@@ -0,0 +1,43 @@
+import { Inject, Injectable } from '@nestjs/common';
+import {
+  INetworkService,
+  NetworkService,
+} from '@/datasources/network/network.service.interface';
+import { ISwapsApi } from '@/domain/interfaces/swaps-api.interface';
+import { ISwapsApiFactory } from '@/domain/interfaces/swaps-api.factory';
+import { HttpErrorFactory } from '@/datasources/errors/http-error-factory';
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { CowSwapApi } from '@/datasources/cow-swap/cowswap-api.service';
+
+@Injectable()
+export class SwapsApiFactory implements ISwapsApiFactory {
+  private readonly apis: Record<string, ISwapsApi> = {};
+
+  constructor(
+    @Inject(NetworkService) private readonly networkService: INetworkService,
+    private readonly httpErrorFactory: HttpErrorFactory,
+    @Inject(IConfigurationService)
+    private readonly configurationService: IConfigurationService,
+  ) {}
+
+  get(chainId: string): ISwapsApi {
+    if (this.apis[chainId]) {
+      return this.apis[chainId];
+    }
+
+    const baseUrl = this.configurationService.get<string>(
+      `swaps.api.${chainId}`,
+    );
+    if (!baseUrl) {
+      throw new Error(
+        `No registered CowSwap API instance for chain ${chainId}`,
+      );
+    }
```

Comment:
Nit: this could be condensed using `getOrThrow`.

Reply:
Initially I was using `getOrThrow` but I wanted to provide a more meaningful message without adding a try/catch for mapping. I've changed it as we are also not dealing with the error right now (we don't need to).

Reply:
Changed in [ddaf881](https://github.com/safe-global/safe-client-gateway/pull/1233/commits/ddaf881310b8a594df12d82f1ddbebe449cb33d5)